### PR TITLE
Implement archive cleanup

### LINKED ISSUE #769: Cached repository archives grow space occupation indefinitevely
Repository archives are only deleted upon explicit request ("Delete all repository archives" from admin dashboard). This results in an indefinite grow of disk space occupation. It may be good to allow for a periodic cleanup instead, maybe based on last access time.

### PR DESCRIPTION
**Pull request by minecrafter**
Fixes #769.

This allows Gitea to automatically purge created archives in order to better control disk space usage. Currently, archives older than 24 hours will be purged every 24 hours.

### CODE REVIEW

File: models/repo.go [hunk 1]
```diff
@@ -1837,6 +1837,58 @@ func DeleteRepositoryArchives() error {
...
+	if err := x.
+		Where("id > 0").
+		Iterate(new(Repository),
+			func(idx int, bean interface{}) error {
+				repo := bean.(*Repository)
+				basePath := filepath.Join(repo.RepoPath(), "archives")
+
+				for _, ty := range []string{"zip", "targz"} {
+					path := filepath.Join(basePath, ty)
+					file, err := os.Open(path)
```
Comment (lunny) [inline models/repo.go]: file is not close

Comment (minecrafter) [reply models/repo.go]: Fixed.

**Comment by lunny** [comment]
LGTM

File: models/repo.go [hunk 2]
```diff
@@ -1837,6 +1837,59 @@ func DeleteRepositoryArchives() error {
...
+					if err != nil {
+						log.Warn("Unable to read directory %s: %v", path, err)
+						return err
+					}
+
+					minimumOldestTime := time.Now().Add(-setting.Cron.ArchiveCleanup.OlderThan)
+					for _, info := range files {
+						if info.ModTime().Before(minimumOldestTime) && !info.IsDir() {
+							toDelete := filepath.Join(path, info.Name())
+							// This is a best-effort purge, so we do not check error codes to confirm removal.
```
Comment (strk) [inline models/repo.go]: I'd still trace-print any error (could be something a sysadmin could fix...)

File: models/repo.go [hunk 3]
```diff
@@ -1837,6 +1837,59 @@ func DeleteRepositoryArchives() error {
...
+	}
+	taskStatusTable.Start(archiveCleanup)
+	defer taskStatusTable.Stop(archiveCleanup)
+
+	log.Trace("Doing: ArchiveCleanup")
+
+	if err := x.
+		Where("id > 0").
+		Iterate(new(Repository),
+			func(idx int, bean interface{}) error {
```
Comment (strk) [inline models/repo.go]: Could this be defined as a standalone function, for better readability ?

**Review by strk — APPROVED** [review]
LGTM - please also update documentation (ini cheatsheet)

**Comment by strk** [comment]
As my review wasn't immediately recognized by the bot, I'll take the chance to ask you to deal with the two comments I've added before convincing the bot :)

**Comment by strk** [comment]
LGTM now - although I don't know how cheatsheet is generated

**Pull request merged**